sklearn/inspection/_plot/tests/test_plot_partial_dependence.py

```diff
@@ -773,11 +845,42 @@ def test_plot_partial_dependence_lines_kw(
     if pd_line_kw is not None and "linestyle" in pd_line_kw:
         assert line.get_linestyle() == pd_line_kw["linestyle"]
     else:
-        assert line.get_linestyle() == "-"
+        assert line.get_linestyle() == "--"
 
     line = disp.lines_[0, 0, 0]
     assert line.get_color() == expected_colors[1]
     if ice_lines_kw is not None and "linestyle" in ice_lines_kw:
         assert line.get_linestyle() == ice_lines_kw["linestyle"]
     else:
         assert line.get_linestyle() == "-"
+
+
+def test_partial_dependence_display_wrong_len_kind(
+    pyplot,
+    clf_diabetes,
+    diabetes,
+):
+    """Check that we raised an error when the `kind` is a list and does not have the
+    right length.
+
+    This bug cannot be trigger apart of creating a `PartialDependenceDisplay` without
+    calling `.from_estimator`.
+    """
+    disp = PartialDependenceDisplay.from_estimator(
+        clf_diabetes,
+        diabetes.data,
+        features=[0, 2],
+        grid_resolution=20,
+        kind="average",
```

Comment:
Why not pass a list with the wrong length directly instead of changing it later?

```suggestion
        kind=["average"],  # len(kind) != len(features)
```